packages/node-cli/src/commands/Command.js

```diff
@@ -38,9 +38,10 @@ class Command {
 		this._runInInteractiveMode = options.runInInteractiveMode;
 		this._interactiveSupport = options.interactiveSupport;
 		this._log = options.log;
+		this._executionEnvironmentContext = options.executionEnvironmentContext;
 
-		this._action = new action(options);
-		this._inputHandler = new inputHandler(options);
+		this._action = new action(options, this._executionEnvironmentContext);
+		this._inputHandler = new inputHandler(options, this._executionEnvironmentContext);
```

Comment:
First of all, great refactor Karl it looks much cleaner than the previous implementation.!!!
The executionEnvironmentContext is included in the options object, and we don't need to add an extra parameter when constructing the Action or InputHandler.
Actually in the BaseAction this second parameter is not used and the executionEnvironmentContet is take from the options. Same comment is valid for InputHandler. I also think that sdkPath can be added as a property in the BaseInputHandler
Snnipet from BaseAction:
```
module.exports = class BaseAction {
	constructor(options) {
		this._projectFolder = options.projectFolder;
		this._commandMetadata = options.commandMetadata;
		this._executionPath = options.executionPath;
		this._runInInteractiveMode = options.runInInteractiveMode;
		this._log = options.log;
		this._sdkPath = options.sdkPath;


		this._sdkExecutor = new SdkExecutor(this._sdkPath);
		this._executionEnvironmentContext = options.executionEnvironmentContext;
		this._sdkExecutor = new SdkExecutor(this._sdkPath, this._executionEnvironmentContext);
	}
```

Reply:
I agree, it's looks much cleaner now.
Did some more updates based on your suggestion.